### PR TITLE
API-4183: Delete unused applications job

### PR DESCRIPTION
**Pull request by cjrowe**

### CODE REVIEW

File: app/uk/gov/hmrc/apiplatformjobs/repository/UnusedApplicationsRepository.scala [hunk 1]
```diff
@@ -58,12 +58,12 @@ class UnusedApplicationsRepository @Inject()(mongo: ReactiveMongoComponent)(impl
       background = true)
   )
 
-  def applicationsByEnvironment(environment: Environment): Future[List[UnusedApplication]] = find("environment" -> environment)
+  def unusedApplications()(implicit environment: Environment): Future[List[UnusedApplication]] = find("environment" -> environment)
```
Comment (mattclark-zerogravit) [inline app/uk/gov/hmrc/apiplatformjobs/repository/UnusedApplicationsRepository.scala]: does environment need to be implicit here? if its used in the query would it make more sense to have it as an argument to the function?

File: app/uk/gov/hmrc/apiplatformjobs/repository/UnusedApplicationsRepository.scala [hunk 2]
```diff
@@ -58,12 +58,12 @@ class UnusedApplicationsRepository @Inject()(mongo: ReactiveMongoComponent)(impl
...
 
-  def applicationsToBeDeleted(environment: Environment, deletionDate: DateTime = DateTime.now): Future[List[UnusedApplication]] =
+  def unusedApplicationsToBeDeleted(deletionDate: DateTime = DateTime.now)(implicit environment: Environment): Future[List[UnusedApplication]] =
```
Comment (mattclark-zerogravit) [inline app/uk/gov/hmrc/apiplatformjobs/repository/UnusedApplicationsRepository.scala]: does environment need to be implicit here? if its used in the query would it make more sense to have it as an argument to the function?

File: app/uk/gov/hmrc/apiplatformjobs/repository/UnusedApplicationsRepository.scala [hunk 3]
```diff
@@ -58,12 +58,12 @@ class UnusedApplicationsRepository @Inject()(mongo: ReactiveMongoComponent)(impl
...
     find("environment" -> environment, "scheduledDeletionDate" -> Json.obj("$lte" -> deletionDate))
 
-  def deleteApplication(environment: Environment, applicationId: UUID): Future[Boolean] =
+  def deleteUnusedApplicationRecord(applicationId: UUID)(implicit environment: Environment): Future[Boolean] =
```
Comment (mattclark-zerogravit) [inline app/uk/gov/hmrc/apiplatformjobs/repository/UnusedApplicationsRepository.scala]: does environment need to be implicit here? if its used in the query would it make more sense to have it as an argument to the function?

File: app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala [hunk 4]
```diff
@@ -136,9 +136,8 @@ class UpdateUnusedProductionApplicationRecordsJob @Inject()(@Named("tpa-producti
...
                                                             mongo: ReactiveMongoComponent)
   extends UpdateUnusedApplicationRecordsJob(
-    Environment.PRODUCTION,
     thirdPartyApplicationConnector,
     thirdPartyDeveloperConnector,
     unusedApplicationsRepository,
     configuration,
-    mongo)
\ No newline at end of file
+    mongo)(Environment.PRODUCTION)
```
Comment (mattclark-zerogravit) [inline app/uk/gov/hmrc/apiplatformjobs/scheduled/UpdateUnusedApplicationRecordsJob.scala]: new line here

**Review by mattclark-zerogravit — COMMENTED** [review]
Question about implicit parameter to repository

**Review by lburgos — APPROVED** [review]

**Pull request merged**